source/transfers/test/TransferHandlerRegistry.js

```diff
@@ -0,0 +1,1022 @@
+    })
+
+    it('Checks balances...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenOMG, 800],
+          [tokenDAI, 100],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenOMG, 200],
+          [tokenDAI, 900],
+        ]),
+        'Bob balances are incorrect'
+      )
+    })
+
+    it('Checks that Bob cannot take the same order again (200 OMG for 50 DAI)', async () => {
+      await reverted(
+        swap(_order, { from: bobAddress }),
+        'ORDER_TAKEN_OR_CANCELLED'
+      )
+    })
+
+    it('Checks balances and approvals...', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenOMG, 800],
+          [tokenDAI, 100],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenOMG, 200],
+          [tokenDAI, 900],
+        ]),
+        'Bob balances are incorrect'
+      )
+      ok(
+        await allowances(aliceAddress, swapAddress, [
+          [tokenOMG, 0],
+          [tokenDAI, 0],
+        ])
+      )
+    })
+
+    it('Checks that Alice cannot trade 200 OMG when allowance is 0', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: aliceAddress,
+          token: tokenOMG.address,
+          amount: 200,
+        },
+        sender: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 10,
+        },
+      })
+
+      await reverted(swap(order, { from: bobAddress }))
+    })
+
+    it('Checks that Bob can not trade more OMG tokens than he holds', async () => {
+      const order = await orders.getOrder({
+        signer: {
+          wallet: bobAddress,
+          token: tokenDAI.address,
+          amount: 1000,
+        },
+        sender: {
+          wallet: aliceAddress,
+        },
+      })
+      await reverted(swap(order, { from: aliceAddress }))
+    })
+
+    it('Checks remaining balances and approvals', async () => {
+      ok(
+        await balances(aliceAddress, [
+          [tokenOMG, 800],
+          [tokenDAI, 100],
+        ]),
+        'Alice balances are incorrect'
+      )
+      ok(
+        await balances(bobAddress, [
+          [tokenOMG, 200],
+          [tokenDAI, 900],
+        ]),
+        'Bob balances are incorrect'
+      )
+      ok(
+        await allowances(aliceAddress, swapAddress, [
+          [tokenOMG, 0],
+          [tokenDAI, 0],
+        ])
+      )
+      ok(
+        await allowances(bobAddress, swapAddress, [
+          [tokenOMG, 0],
+          [tokenDAI, 900],
+        ])
+      )
+    })
+  })
+
+  describe('Deploying NFT tokens...', async () => {
+    it('Deployed test contract "ConcertTicket"', async () => {
+      tokenTicket = await NonFungibleToken.new()
+    })
+
+    it('Deployed test contract "Collectible"', async () => {
+      tokenKitty = await NonFungibleToken.new()
```

Comment:
Maybe I'm misunderstanding something here... here's my understanding:
- we have a separate kitty core handler due to the fact that crypto kitties are transferred in a different way to standard ERC721 tokens (and don't have a safeTransferFrom function)
- This means the ERC721 handler should be used for standard NFTs, but the kitty handler should be used for crypto kitties.
- These tests are using the CK Handler using a standard NFT, and so it's never tested whether the CK Handler works with the CK token contract, which is its purpose.

If this is correct, maybe the tests should use the CK token contract we have in the repo instead of the NFT contract?

Reply:
tldr, added new AdaptedKittyERC721 so that this contract is mintable, lacks a safetransferFrom and uses custom event names to ensure Kitty transfer is occurring.